app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,24 @@
+class SessionsController < ApplicationController
+
+  def login_form
+  end
+
+  def login
+    user = User.find_by(username: params[:username])
```

Comment:
one small bug-- the `username` field in the login_form.html.erb is optional, so you can "log in" with a blank username :) This doesn't really affect the site's experience but yeah!